src/test/java/dev/openfeature/sdk/FlagEvaluationSpecTest.java

```diff
@@ -99,7 +99,7 @@ void getApiInstance() {
 
     @Specification(number="2.4.5", text="The provider SHOULD indicate an error if flag resolution is attempted before the provider is ready.")
     @Test void shouldReturnNotReadyIfNotInitialized() {
-        FeatureProvider provider = new InMemoryProvider(new HashMap<>()) ;
+        FeatureProvider provider = new TestEventsProvider(100);
```

Comment:
This test was flaky, since after your changes, it was possible that the provider was initialized before line 105 ran. This adds an artificial delay to init to make sure.

Notice that I'm NOT using `setProviderAndWait` here because we explicitly DONT want the provider to be ready for this test.